libsolidity/interface/CompilerStack.cpp

```diff
@@ -508,23 +508,35 @@ string CompilerStack::applyRemapping(string const& _path, string const& _context
 		return std::equal(_a.begin(), _a.end(), _b.begin());
 	};
 
+	using filepath = boost::filesystem::path;
 	size_t longestPrefix = 0;
+	size_t longestContext = 0;
 	string longestPrefixTarget;
+
 	for (auto const& redir: m_remappings)
 	{
-		// Skip if we already have a closer match.
-		if (longestPrefix > 0 && redir.prefix.length() <= longestPrefix)
+		filepath redirContext(redir.context);
+		filepath redirPrefix(redir.prefix);
+		string contextFileString = redirContext.generic_string();
```

Comment:
converts it directly into exact filepath format if the string wasn't already in a proper "filepath" format and makes the comparisons consistent. Say I put in myDir and myDir/ you'd agree that from a directory standpoint they are exactly the same, but from a string perspective, clearly they are not. This resolves that.